From 07d7ed26a59bf834d38cae369670b613e4cea2f2 Mon Sep 17 00:00:00 2001
From: Mohammed Naser <mnaser@vexxhost.com>
Date: Sun, 19 Feb 2023 18:47:11 +0000
Subject: [PATCH] fix(swift): allow object creation

It is currently not possible to create any new objects inside
Swift as you'd always get an error that the new object name
already exists all the time.

This fixes that issue and enables you to upload files again.

Related-Change-Id: Idfb90a327e94ccaa3263aaaad1d6d52fa46312f4
Change-Id: Ib2f1aaae87820e8d5c4a6c9b01b6a6e9a5552952
---

diff --git a/openstack_dashboard/static/app/core/openstack-service-api/swift.service.js b/openstack_dashboard/static/app/core/openstack-service-api/swift.service.js
index b2b170e..daafd8e 100644
--- a/openstack_dashboard/static/app/core/openstack-service-api/swift.service.js
+++ b/openstack_dashboard/static/app/core/openstack-service-api/swift.service.js
@@ -296,8 +296,7 @@
         service.getObjectURL(container, objectName, 'metadata')
       );
       if (ignoreError) {
-        // provide a noop error handler so the error is ignored
-        return promise.catch(angular.noop);
+        return promise;
       }
       return promise.catch(function onError() {
         toastService.add('error', gettext('Unable to get details of the object.'));
diff --git a/openstack_dashboard/static/app/core/openstack-service-api/swift.service.spec.js b/openstack_dashboard/static/app/core/openstack-service-api/swift.service.spec.js
index 40ec000..06a6490 100644
--- a/openstack_dashboard/static/app/core/openstack-service-api/swift.service.spec.js
+++ b/openstack_dashboard/static/app/core/openstack-service-api/swift.service.spec.js
@@ -256,7 +256,6 @@
       spyOn(promise, 'catch');
       spyOn(toastService, 'add');
       service.getObjectDetails('spam', 'ham', true);
-      expect(promise.catch).toHaveBeenCalledWith(angular.noop);
       expect(toastService.add).not.toHaveBeenCalled();
     });
 
